web-server/sgx-wallet-test/enclave/src/ported/test_kv_store_fs.rs

```diff
@@ -0,0 +1,33 @@
+use std::prelude::v1::*;
+
+use proptest::prelude::*;
+use sgx_wallet_impl::ported::kv_store::fs::{decode_from_fs_safe, encode_to_fs_safe};
+use sgx_wallet_impl::ported::kv_store::Key;
+
+/// [`encode_to_fs_safe`] encodes to filesystem-safe, and [`decode_from_fs_safe`] round-trips.
+pub(crate) fn prop_fs_safe_roundtrip() {
+    prop_fs_safe_roundtrip_wrapper()
+}
+
+proptest! {
+    fn prop_fs_safe_roundtrip_wrapper(key: Box<Key>) {
```

Comment:
Can we use the closure style invocation here as well to stay consistent inside of the project or will it not work for this case?

Reply:
It won't work for this case currently: the closure-style invocation doesn't support specifying the return type, which is necessary for using `prop_assert!` (which is a hard requirement for the enclave-based tests, because `assert!` panics are actually fatal / not caught by proptest).

(I have some in-progress project documentation for this, which I'll follow-up with at some point.)